src/utilities/Cd.ts

```diff
@@ -60,8 +60,9 @@ export default class Cd extends BaseUtility {
 }
 
 export function setUrlLocation({ location }: { location: string }): void {
-	// const currentUrl = window.location.href;
-	// const newUrl =
-	// 	currentUrl.substring(0, currentUrl.lastIndexOf('/') + 1) + location;
-	// window.history.pushState({ path: newUrl }, '', newUrl);
+	const currentUrl = window.location.href;
+	const tagIndex = currentUrl.lastIndexOf('#') === -1 ? currentUrl.lastIndexOf('/') : currentUrl.lastIndexOf('#') 
+	const newUrl =
+		currentUrl.substring(0, tagIndex) + '#' + location;
```

Comment:
Thank you so much for figuring out this solution!

There is a more idiomatic way of doing this. Instead of doing all this string manipulation on `window.location.href`, you can make use of the `window.location.hash` property.

Probably you could just do `window.location.hash = location`. If you don't feel like implementing that, let me know and I will merge this pull request anyways!

Reply:
Done.